KubeArmor/monitor/logUpdate.go

```diff
@@ -301,6 +301,36 @@ func (mon *SystemMonitor) UpdateLogs() {
 				log.Resource = ""
 				log.Data = "syscall=" + getSyscallName(int32(msg.ContextSys.EventID)) + " fd=" + fd
 
+			case SysPtrace:
+				if len(msg.ContextArgs) != 2 {
+					continue
+				}
+
+				var request string
+				var pid string
+
+				if val, ok := msg.ContextArgs[0].(string); ok {
+
+					request = val
+				}
+				var traceePid uint32
+				if val, ok := msg.ContextArgs[1].(int32); ok {
+					traceePid = uint32(val)
+					pid = strconv.Itoa(int(val))
+				}
+
+				if msg.ContainerID == "" {
+					log.Resource = mon.GetCommand(msg.ContainerID, traceePid)
+
+				} else {
+					ActivePidProcMap := *(mon.ActivePidProcMap)
+					hostPid := ActivePidProcMap[msg.ContainerID][traceePid]
+					log.Resource = mon.GetCommand(msg.ContainerID, hostPid)
```

Comment:
Can you document why we need to do this? what is traceePid? why do we keep a seperate map to track it?

Reply:
tracee pid is the pid of the process that is being traced or the tracer process is attached to. the map is to keep track of pids per container basis as the argument passed with ptrace syscall will give pid number as per container namespace.

Reply:
we've moved from this logic, now using lsm-hook instead of maintaining a separate map.